core/src/main/scala/spinal/core/AFix.scala

```diff
@@ -1178,11 +1222,32 @@ class AFix(val maxRaw: BigInt, val minRaw: BigInt, val exp: Int) extends MultiDa
   }
 }
 
-
+object AF {
+  def apply(value: BigDecimal, integerWidth: BitCount, fractionWidth: BitCount, signed: Boolean): AFix = {
+    val ret = if (signed) AFix.SQ(integerWidth, fractionWidth) else AFix.UQ(integerWidth, fractionWidth)
+    val tmp = cloneOf(ret)
+    tmp := value
+    ret.raw.assignFromBits(tmp.raw)
+    ret
+  }
+}
 
 class TagAFixTruncated(val saturation: Boolean,
                        val overflow  : Boolean,
-                       val rounding  : RoundType) extends SpinalTag{
+                       val rounding  : RoundType
+                       ) extends SpinalTag{
   override def duplicative = true
   override def canSymplifyHost: Boolean = true
-}
\ No newline at end of file
+}
+
+object AFixTruncatedScope extends ScopeProperty[TagAFixTruncated] {
+  override def default: TagAFixTruncated = new TagAFixTruncated(true, false, RoundType.ROUNDTOINF)
```

Comment:
Maybe the default should be different, and aim at the most raw way : 
saturation = false,
overflow = false, 
rounding = RoundType.FLOOR

Reply:
Anyway in general, doesn't it seems to be dangerious to have AFixTruncatedScope  ?
For instance, you developpe some hardware with a given default, and instanciate some sub block which was designed with the idea in mind that AFixTruncatedScope isn't altered => broken hardware.

Also, one way to see things is that saturation / overflow / rounding should be explicitly asked by the user, else missmatched would throw you an error.

Reply:
hmm i may have missunderstood the usage of AFixTruncatedScope 
So, it is only to provide some sort of implicit parameters to some user called function right ?

Reply:
Yes, `AFixTruncationScope` is to provide some implicit parameter to only rounding functions. Users can always pass an explicit parameter to o override it. You can check out the `getTrunc` function. 
Also, the scope does not influence the assignment. You have to add a `TagAFixTruncated` to do truncation when doing an assignment. 

Speaking of the default rounding options, maybe we can add them to SpinalConfig to allow users to define the default?

Reply:
Ahhhhh, ok, i think i was confused in part because it reuse the TagAFixTruncated class, while it has kind of nothing to do the the original usage of the TagAFixTruncated.

So, i think that's good to merge, i will probably just create a separated class TagAFixTruncated -> RoundingDefault or something in that kind.

All good on your side ?

Reply:
> Speaking of the default rounding options, maybe we can add them to SpinalConfig to allow users to define the default?

I would say it isn't necessary, i mean, the concept of ScopeProperty is that you could set them where you want, including outside the Spinal context (before SpinalVerilog(xxx)). So SpinalConfig does not have to be coupled with it.

Reply:
> Also, still, it can be quite dangerous to change the default implicit rounding of a given design, especially when doing the integration of multiple things

If users wants to avoid changing or misusing the default rounding, they can wrap their code into a afix truncation scope or pass rounding parameters explicitly.